fe/src/main/java/org/apache/doris/load/routineload/KafkaRoutineLoadJob.java

```diff
@@ -276,17 +301,9 @@ protected String getStatistic() {
         return gson.toJson(summary);
     }
 
-    private List<Integer> getAllKafkaPartitions() throws LoadException {
-        List<Integer> result = new ArrayList<>();
-        try {
-            List<PartitionInfo> partitionList = consumer.partitionsFor(topic);
-            for (PartitionInfo partitionInfo : partitionList) {
-                result.add(partitionInfo.partition());
-            }
-        } catch (Exception e) {
-            throw new LoadException("failed to get partitions for topic: " + topic + ". " + e.getMessage());
-        }
-        return result;
+    private List<Integer> getAllKafkaPartitions() throws UserException {
+        convertCustomProperties();
+        return KafkaUtil.getAllKafkaPartitions(brokerList, topic, convertedCustomProperties);
```

Comment:
If the convertedCustomProperties is incorrect, the job will not be cancelled? The connection timeout and certification failed need to be distinguish.

Reply:
Yes, if convertedCustomProperties is incorrect, the job will be paused.
And it is not easy to distinguish timeout and other failure. for example, if ssl authentication failed, kafka client on BE will retry again and again , while in FE, it looks like timeout.

Reply:
Of cause, you remind me that convertedCustomProperties should be re-created when routine load job transfer from PAUSE to NEED_SCHEDULE, because some files may be re-created anytime.